src/Volume.ts

```diff
@@ -200,14 +209,27 @@ export default class Volume {
     this.normRegionOffset = subregionOffset.clone().divide(volumeSize);
   }
 
+  updateRequiredData(required: Partial<LoadSpec>) {
+    this.loadSpecRequired = { ...this.loadSpecRequired, ...required };
+    // if newly required data is not currently contained in this volume...
+    if (
+      this.loadSpecRequired.time !== this.loadSpec.time ||
+      this.loadSpecRequired.scene !== this.loadSpec.scene ||
+      !this.loadSpec.subregion.containsBox(this.loadSpecRequired.subregion)
+    ) {
+      // ...clone `loadSpecRequired` into `loadSpec` and load
+      this.loadSpec = { ...this.loadSpecRequired, subregion: this.loadSpecRequired.subregion.clone() };
+      this.loader?.loadVolumeData(this);
+    }
+  }
+
   // we calculate the physical size of the volume (voxels*pixel_size)
   // and then normalize to the max physical dimension
   setVoxelSize(size: Vector3): void {
     // only set the data if it is > 0.  zero is not an allowed value.
-    size.clampScalar(0, Infinity);
-    size.x = size.x || 1.0;
-    size.y = size.y || 1.0;
-    size.z = size.z || 1.0;
+    size.x = size.x > 0 ? size.x : 1.0;
+    size.y = size.y > 0 ? size.y : 1.0;
+    size.z = size.z > 0 ? size.z : 1.0;
```

Comment:
What does this do if it receives a Vector3 with infinite values?

Reply:
The behavior here is hopefully identical to the previous code, just a bit more efficient. This code will leave `+Infinity` alone and turn `-Infinity` into 1.